cmd/livepeer/livepeer.go

```diff
@@ -393,9 +397,24 @@ func main() {
 		}
 
 		if n.NodeType == core.BroadcasterNode {
+			ev, _ := new(big.Rat).SetString(*maxTicketEV)
+			if ev == nil {
+				panic(fmt.Errorf("-maxTicketEV must be a valid rational number, but %v provided. Restart the node with a valid value for -maxTicketEV", *maxTicketEV))
+			}
+
+			if ev.Cmp(big.NewRat(0, 1)) < 0 {
+				panic(fmt.Errorf("-maxTicketEV must not be negative, but %v provided. Restart the node with a valid value for -maxTicketEV", *maxTicketEV))
+			}
+
+			if *depositMultiplier <= 0 {
+				panic(fmt.Errorf("-depositMultiplier must be greater than 0, but %v provided. Restart the node with a valid value for -depositMultiplier", *depositMultiplier))
+			}
+
 			// TODO: Initialize Sender with an implementation
 			// of RoundsManager that reads from a cache
-			n.Sender = pm.NewSender(n.Eth, n.Eth)
+			// TODO: Initialize Sender with an implementation
+			// of SenderManager that reads from a cache
+			n.Sender = pm.NewSender(n.Eth, n.Eth, n.Eth, ev, *depositMultiplier)
```

Comment:
This argument list is starting to look weirder and weirder by the day to me 😅 . But I can see why it's done this way, the EthClient implements the respective interfaces in the constructor parameter list. I'm just wondering if there isn't a better way to do this such as dividing up the EthClient into those structs that implement those respective interfaces. Rather than having their implementation defined on the client struct directly. Altough introducing another level of wrappers might not be a good idea here. 

I'm not sure, it just looks odd.

Reply:
Yeah I hear you. This should eventually change though as each of those interfaces should be implemented by structs that cache values instead of executing a RPC request per method call.

Reply:
I can live with that 👍